src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -202,16 +208,27 @@ export class TerminalService implements ITerminalService {
 			}
 		);
 
+		this._forwardInstanceHostEvents(this._terminalGroupService);
+		this._forwardInstanceHostEvents(this._terminalEditorService);
+		this._terminalGroupService.onDidChangeActiveGroup(this._onActiveGroupChanged.fire, this._onActiveGroupChanged);
+		this._terminalGroupService.onPanelOrientationChanged(this._onPanelOrientationChanged.fire, this.onPanelOrientationChanged);
 
 		// the below avoids having to poll routinely.
 		// we update detected profiles when an instance is created so that,
 		// for example, we detect if you've installed a pwsh
 		this.onInstanceCreated(() => this._refreshAvailableProfiles());
-		this.onGroupDisposed(group => this._removeGroup(group));
-		this.onInstancesChanged(() => this._terminalCountContextKey.set(this._terminalInstances.length));
-		this.onGroupsChanged(() => this._terminalGroupCountContextKey.set(this._terminalGroups.length));
+		this.onDidChangeInstances(() => this._terminalCountContextKey.set(this.instances.length));
 		this.onInstanceLinksReady(instance => this._setInstanceLinkProviders(instance));
 
+		// Hide the panel if there are no more instances, provided that VS Code is not shutting
+		// down. When shutting down the panel is locked in place so that it is restored upon next
+		// launch.
+		this._terminalGroupService.onDidChangeActiveInstance(instance => {
+			if (!instance && !this._isShuttingDown) {
+				this.hidePanel();
```

Comment:
don't think `!instance` would ever be true bc `terminalGroupService `doesn't allow setting the `activeInstance `to undefined
https://github.com/microsoft/vscode/blob/fbf73bd8e4e0f750a0143b1b985d12a3530a8362/src/vs/workbench/contrib/terminal/browser/terminalGroupService.ts#L76-L81

Reply:
It can be undefined if there are no groups for example, this is not managed only by the group service though with https://github.com/microsoft/vscode/pull/126674/commits/f97906dfb12086fd68285c9e7fcf067af5086959